### PR TITLE
(Bug) #662 User don't get the verification email during wallet creation

### PR DESCRIPTION
**Pull request by StanislavShevchenko**
# Description

MutatikID was created incorrectly, due to incorrect input data

Fixes [User don't get the verification email during wallet creation#662](https://app.zenhub.com/workspaces/gooddollar-5d50833888a83c6880d3e345/issues/gooddollar/gooddapp/662)

# How Has This Been Tested?

Register and go to verification email screen

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request ( for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

**Comment by StanislavShevchenko** [comment]
> we need changes to prevent duplicate creation of mauticId @StanislavShevchenko

done

File: src/server/verification/verificationAPI.js [hunk 1]
```diff
@@ -247,27 +247,33 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
     wrapAsync(async (req, res, next) => {
       const log = req.log.child({ from: 'verificationAPI - verify/sendemail' })
       const { user, body } = req
+      let userData
       // log.info({ user, body })
       //merge user details for use by mautic
-      let userRec: UserRecord = _.defaults(body.user, user)
+      try {
+        userData = user.toJSON()
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js]: this needs to be done at the source.
this issue will repeat everywhere we use results from mongodb

Comment (StanislavShevchenko) [reply src/server/verification/verificationAPI.js]: done

Comment (sirpy) [reply src/server/verification/verificationAPI.js]: need to change it at the mongo adapter level

**Pull request merged**